app/assets/javascripts/float-kit/addon/components/d-default-toast.gjs

```diff
@@ -0,0 +1,61 @@
+import Component from "@glimmer/component";
+import DButton from "discourse/components/d-button";
+import icon from "discourse-common/helpers/d-icon";
+import { concat, fn } from "@ember/helper";
+import concatClass from "discourse/helpers/concat-class";
+import or from "truth-helpers/helpers/or";
+import { action } from "@ember/object";
+
+export default class DDefaultToast extends Component {
+  <template>
+    <div
+      class={{concatClass
+        "d-default-toast"
+        (concat "-" (or @model.theme "default"))
+      }}
+      ...attributes
+    >
+      {{#if @model.icon}}
+        <div class="d-default-toast__icon-container">
+          {{icon @model.icon}}
+        </div>
+      {{/if}}
+      <div class="d-default-toast__main-container">
+        <div class="d-default-toast__texts">
+          {{#if @model.title}}
+            <div class="d-default-toast__title">
+              {{@model.title}}
+            </div>
+          {{/if}}
+          {{#if @model.message}}
+            <div class="d-default-toast__message">
+              {{@model.message}}
+            </div>
+          {{/if}}
+        </div>
+
+        {{#if @model.actions}}
+          <div class="d-default-toast__actions">
+            {{#each @model.actions as |toastAction|}}
+              <DButton
+                @icon={{toastAction.icon}}
+                @translatedLabel={{toastAction.label}}
+                @action={{fn this.handleAction toastAction.action}}
```

Comment:
If you like, we could remove the intermediate action like
```suggestion
                @action={{fn toastAction.action (hash model=@model close=@close)}}
```

Reply:
Yes sure :+1:

Reply:
and then next you will ask for template only :p

Reply:
done